startup_runners/outer_startup_runner.c

```
@@ -35,7 +35,7 @@ main(int argc, char **argv)
 {
     int result, num_per_core_data;
 
-    if (argc != 2) {
+    if (argc < 2) {  // krun passs more args, which are ignored here.
```

Typo: "passs".

Also the comment doesn't make sense to me: if Krun passes more than the expected number of args, this program will raise an error (rather than ignoring them).

---

Typo yes.

Disagree with the second statement though. *less than*.

---

The comment still doesn't make sense. If Krun passes (say) 5 arguments, the program will raise an error. Something doesn't add up to me.

---

No it won't. If we pass 5 args, argc will be 6. `(6 < 2)` is false.

---

Ah, I see what you mean. Can we simplify the comment to something like "Ensure we get the arguments we need, but we're OK with extra arguments being passed (such as by Krun)". [OK, that's a rubbish comment, but you get the idea.]

---

Ah, I get it now.